libraries/rtos/rtos/Thread.cpp

```diff
@@ -41,6 +41,11 @@ Thread::Thread(void (*task)(void const *argument), void *argument,
             error("Error allocating the stack memory\n");
         _dynamic_stack = true;
     }
+    
+    //Fill the stack with a magic word for maximum usage checking
+    for (int i = 0; i < (stack_size / sizeof(uint32_t)); i++) {
+        _thread_def.stack_pointer[i] = 0xE25A2EA5;
```

Comment:
Note that `0xE25A2EA5` is the same magic word defined in *rt_HAL_CM.h* for stack-overflow checking.